drivers/volume/kdmp/kdmp.go

```diff
@@ -0,0 +1,300 @@
+package kdmp
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	snapv1 "github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis/crd/v1"
+	snapshotVolume "github.com/kubernetes-incubator/external-storage/snapshot/pkg/volume"
+	storkvolume "github.com/libopenstorage/stork/drivers/volume"
+	storkapi "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/libopenstorage/stork/pkg/errors"
+	"github.com/libopenstorage/stork/pkg/log"
+	kdmpapi "github.com/portworx/kdmp/pkg/apis/kdmp/v1alpha1"
+	"github.com/portworx/kdmp/pkg/drivers"
+	"github.com/portworx/sched-ops/k8s/core"
+	kdmpShedOps "github.com/portworx/sched-ops/k8s/kdmp"
+	"github.com/portworx/sched-ops/k8s/storage"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8shelper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+)
+
+const (
+	// driverName is the name of the aws driver implementation
+	driverName    = "kdmp"
+	proxyEndpoint = "proxy_endpoint"
+	proxyPath     = "proxy_nfs_exportpath"
+	// StorkAPIVersion current api version supported by stork
+	StorkAPIVersion = "stork.libopenstorage.org/v1alpha1"
+)
+
+type kdmp struct {
+	storkvolume.ClusterPairNotSupported
+	storkvolume.MigrationNotSupported
+	storkvolume.GroupSnapshotNotSupported
+	storkvolume.ClusterDomainsNotSupported
+	storkvolume.CloneNotSupported
+	storkvolume.SnapshotRestoreNotSupported
+}
+
+func (k *kdmp) Init(_ interface{}) error {
+	return nil
+}
+
+func (k *kdmp) String() string {
+	return driverName
+}
+
+func (k *kdmp) Stop() error {
+	return nil
+}
+
+func (k *kdmp) OwnsPVC(coreOps core.Ops, pvc *v1.PersistentVolumeClaim) bool {
+	// Try to get info from the PV since storage class could be deleted
+	pv, err := coreOps.GetPersistentVolume(pvc.Spec.VolumeName)
+	if err != nil {
+		log.PVCLog(pvc).Warnf("error getting pv %v for pvc %v: %v", pvc.Spec.VolumeName, pvc.Name, err)
+		return false
+	}
+	// for px proxy volume use kdmp by default
+	storageClassName := k8shelper.GetPersistentVolumeClaimClass(pvc)
```

Comment:
Would there be any case wherein `storageClassName` is empty and if so, do we need to handle this?

Reply:
yes it can be , in that case it will be handled by looking at pv object